sql/moz-fx-data-shared-prod/telemetry_derived/hcm_clients_v1/query.sql

```diff
@@ -0,0 +1,34 @@
+SELECT
+  DATE(main.submission_timestamp) AS submission_date,
+  main.normalized_channel AS channel,
+  main.environment.system.os.name AS os,
+  COUNT(DISTINCT client_id) AS num_clients,
+FROM
+  `moz-fx-data-shared-prod`.telemetry_stable.main_v5 AS main
+CROSS JOIN
+  UNNEST(main.payload.processes.parent.keyed_scalars.a11y_theme) AS a11y_theme
+WHERE
+  {% if is_init() %}
+    DATE(main.submission_timestamp) >= '2021-01-01'
+  {% else %}
+    DATE(main.submission_timestamp) = @submission_date
+  {% endif %}
+  /*
+  a11y_theme.value is TRUE when an OS high contrast mode (HCM) is present and FALSE otherwise,
+
+  a11y_theme.key is derived from the config pref `browser.display.document_color_use" which is has the
+  following values / meanings:
+  * "always" = always use document colors, regardless of OS HCM or colors chosen in the colors
+    dialog (about:preferences > manage colors)
+  * "never" = never use document colors (this is equivalent to "always use HCM colors")
+  * "default" = use document colors when OS HCM is off, and use OS HCM colors when OS HCM is on.
+
+  This condition selects users who have OS HCM on and who have not forced document colors everywhere
+  (ie. the OS HCM colors should propagate to web content).
+  */
+  AND a11y_theme.value IS TRUE
+  AND a11y_theme.key != "always"
```

Comment:
@MReschenberg I just noticed of the 5 dashboards, only the generic HCM clients one uses the above logic, while the other 4 use this logic:

```sql
(n.key = "never" OR (n.key != "always" AND n.value = TRUE))
```

can you advise on which logic is preferred?

Reply:
Ah sorry this is kinda confusing: they're measuring different things.

This logic: `(n.key = "never" OR (n.key != "always" AND n.value = TRUE))` tracks clients who have either:
 - HCM colors forced everywhere OR
 - OS HCM enabled AND document colors _not_ forced (ie. OS HCM is never being overridden by the use_document_colors preference)

The logic: `a11y_theme.value IS TRUE AND a11y_theme.key != "always"` tracks clients who both:
- have OS HCM enabled AND
- have _not_ forced document colors
(ie. the last part of the first query)

The former allows us to count clients who are overriding document colors without using their OS HCM theme -- they're doing it in Firefox directly. The latter requires clients have an OS HCM enabled to be counted. 

These are two different kinds of users, and we want to be able to measure them each. The generic dashboard should probably be renamed to "OS HCM clients" instead of "HCM clients" if it isn't already named that way 😁

Thanks for your patience, lemme know if you have any other questions.